pkg/plugin/plugin.go

```diff
@@ -20,53 +24,58 @@ type TLSFiles struct {
 	CLIENT    string `json:"client"`
 }
 
-func readTlsFile() (TLSFiles, error){
+func readTlsFile() (TLSFiles, error) {
 	home, err := os.UserHomeDir()
 	if err != nil {
-	  return TLSFiles{}, err
+		return TLSFiles{}, err
 	}
 	clientKey, err := os.ReadFile(home + "/.lightning/testnet/client-key.pem")
 	ca, err := os.ReadFile(home + ".lightning/testnet/ca.pem")
 	client, err := os.ReadFile(home + ".lightning/testnet/client.pem")
-	if err != nil{
-	  return TLSFiles{}, err
+	if err != nil {
+		return TLSFiles{}, err
 	}
 	files := TLSFiles{CA: string(ca), CLIENTKEY: string(clientKey), CLIENT: string(client)}
 	return files, nil
 }
 
-func startServer() {
-	r := gin.Default()
-	r.GET("/ping", func(c *gin.Context) {
+type StartServer[T PluginState] struct{}
+
+func (instance *StartServer[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]string) (map[string]string, error) {
+	if plugin.State.Server != nil {
+		return map[string]string{"message": "Server is already running"}, nil
+	}
+	plugin.State.Server = gin.Default()
+
+	plugin.State.Server.GET("/ping", func(c *gin.Context) {
 		c.JSON(http.StatusOK, gin.H{
 			"message": "pong",
 		})
 	})
 
-	r.POST("/tls", func(c *gin.Context) {
+	plugin.State.Server.POST("/tls", func(c *gin.Context) {
 		var login LOGIN
 		c.BindJSON(&login)
-		if os.Getenv("password") != login.PASSWORD {
+		if plugin.State.Password != login.PASSWORD {
 			c.JSON(200, gin.H{"status": "Access Denied"})
 		}
 		data, _ := readTlsFile()
 		c.JSON(200, gin.H{"data": &data})
 	})
 
-	r.Run() // listen and serve on 0.0.0.0:8080 (for windows "localhost:8080")
+	plugin.State.Server.Run()
+	return map[string]string{"message": "Server up and running,/ listen and serve on 0.0.0.0:8080 "}, nil
 }
 
 type SetPassword[T PluginState] struct{}
 
-func (instance *SetPassword[T]) Call(plugin *plugin.Plugin[T], request map[string]string) (map[string]string, error) {
-	m := make(map[string]string)
-	m["password"] = request["password"]
-	os.Setenv("password", m["password"])
-	return m, nil
+func (instance *SetPassword[T]) Call(plugin *plugin.Plugin[PluginState], request map[string]string) (map[string]string, error) {
+	// m := make(map[string]string)
+	// m["password"] = request["password"]
+	plugin.State.Password = request["password"]
+	return map[string]string{"message": "Password set"}, nil
```

Comment:
```suggestion
	return map[string]string{"message": "Password set", "password": plugin.State.Password}, nil
```

This is not super safe, but we will go to store a rune in the future so we can return it, so you can share the password with the people that you want